app/channel.go

```diff
@@ -1419,7 +1419,10 @@ func (a *App) postJoinTeamMessage(user *model.User, channel *model.Channel) *mod
 }
 
 func (a *App) LeaveChannel(channelId string, userId string) *model.AppError {
-	sc := a.Srv.Store.Channel().Get(channelId, true)
+	channel, errCh := a.Srv.Store.Channel().Get(channelId, true)
```

Comment:
Here we should be using the Async approach, see the example right after your code (`uc` channel creation and the gorouting right after, you have an example in the example PR linked in the issue.